lib/srv/server/ec2_watcher.go

```diff
@@ -500,6 +509,73 @@ func (f *ec2InstanceFetcher) matcherRegions(ctx context.Context, awsOpts []awsco
 	return enabledRegions, nil
 }
 
+func (f *ec2InstanceFetcher) fetchAccountIDsUnderOrganization(ctx context.Context) ([]string, error) {
+	awsOpts := []awsconfig.OptionsFn{
+		awsconfig.WithCredentialsMaybeIntegration(awsconfig.IntegrationMetadata{Name: f.Matcher.Integration}),
+	}
+
+	orgsClient, err := f.AWSOrganizationsGetter(ctx, awsOpts...)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var organizationID string
+	var includeOUs []string
+	var excludeOUs []string
+	if f.Matcher.Organization != nil {
+		organizationID = f.Matcher.Organization.OrganizationID
+		if f.Matcher.Organization.OrganizationalUnits != nil {
+			includeOUs = f.Matcher.Organization.OrganizationalUnits.Include
+			excludeOUs = f.Matcher.Organization.OrganizationalUnits.Exclude
+		}
+	}
+
+	accountIDs, err := organizations.MatchingAccounts(ctx, orgsClient, organizations.MatchingAccountsFilter{
+		IncludeOUs:     includeOUs,
+		ExcludeOUs:     excludeOUs,
+		OrganizationID: organizationID,
+	})
+	if err != nil {
+		convertedErr := libcloudaws.ConvertRequestFailureError(err)
+		if trace.IsAccessDenied(convertedErr) {
+			return nil, trace.BadParameter("discovering instances under an organization requires the following permissions: [%s], add those to the IAM Role used by the Discovery Service", strings.Join(organizations.RequiredAPIs(), ", "))
+		}
+
+		return nil, trace.Wrap(convertedErr)
+	}
+
+	return accountIDs, nil
+}
+
+// allAssumeRoles returns a list of all the AWS Assume Roles that must be assumed.
+// There's a special case when there is no Role to Assume, in this case an empty string is returned.
+// In this situation no AssumeRole should be passed to the AWS client.
+func (f *ec2InstanceFetcher) allAssumeRoles(ctx context.Context) ([]string, error) {
+	if !f.Matcher.HasOrganizationMatcher() {
+		return []string{f.Matcher.AssumeRole.RoleARN}, nil
```

Comment:
will this panic if `AssumeRole` is nil? can it be nil here?

Reply:
I don't think it can be nil, but I'll add a check here, thank you.